src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.ts

```diff
@@ -0,0 +1,388 @@
+        private submissionVersionService: SubmissionVersionService,
+        private datePipe: ArtemisDatePipe,
+    ) {}
+
+    ngOnInit(): void {
+        this.activatedRoute.data.subscribe(({ studentExam: studentExamWithGrade }) => {
+            this.studentExam = studentExamWithGrade.studentExam;
+            this.courseId = this.studentExam.exam!.course!.id!;
+        });
+        this.exerciseIndex = 0;
+        this.pageComponentVisited = new Array(this.studentExam.exercises!.length).fill(false);
+        this.retrieveSubmissionDataAndTimeStamps().subscribe((results) => {
+            const allSubmissions = results.flat();
+            allSubmissions.forEach((result) => {
+                //workaround because instanceof does not work.
+                if (this.isSubmissionVersion(result)) {
+                    const submissionVersion = result as SubmissionVersion;
+                    this.submissionVersions.push(submissionVersion);
+                    this.submissionTimeStamps.push(submissionVersion.createdDate);
+                } else if (this.isFileUploadSubmission(result)) {
+                    const fileUploadSubmission = result as FileUploadSubmission;
+                    this.fileUploadSubmissions.push(fileUploadSubmission);
+                    this.submissionTimeStamps.push(fileUploadSubmission.submissionDate!);
+                } else {
+                    const programmingSubmission = result as ProgrammingSubmission;
+                    this.programmingSubmissions.push(programmingSubmission);
+                    this.submissionTimeStamps.push(programmingSubmission.submissionDate!);
+                }
+            });
+            this.sortTimeStamps();
+            this.setupRangeSlider();
+            const firstSubmission = this.findFirstSubmission();
+            this.currentSubmission = firstSubmission;
+            this.exerciseIndex = this.findExerciseIndex(firstSubmission!);
+            this.examNavigationBarComponent.changePage(false, this.exerciseIndex, false, firstSubmission);
+        });
+    }
+
+    ngAfterViewInit(): void {
+        this.changesSubscription = this.currentPageComponents.changes.subscribe(() => {
+            this.updateSubmissionOrSubmissionVersionInView();
+        });
+    }
+
+    private updateProgrammingExerciseView() {
+        const activeProgrammingComponent = this.activePageComponent as ProgrammingExamSubmissionComponent;
+        if (activeProgrammingComponent) {
+            activeProgrammingComponent!.studentParticipation.submissions![0] = this.currentSubmission as ProgrammingSubmission;
+        }
+    }
+
+    private setupRangeSlider() {
+        this.selectedTimestamp = this.submissionTimeStamps[0]?.toDate().getTime() ?? 0;
+        const newOptions: Options = Object.assign({}, this.options);
+        newOptions.stepsArray = this.submissionTimeStamps.map((date) => {
+            return {
+                value: date.toDate().getTime(),
+            };
+        });
```

Comment:
```suggestion
        newOptions.stepsArray = this.submissionTimeStamps.map((date) => ({ value: date.toDate().getTime() }));
```
optional, matter of taste